worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -499,16 +500,18 @@ private[deploy] class Controller(
       false)
 
     val future =
-      if (primaryFuture != null && replicaFuture != null) {
-        CompletableFuture.allOf(primaryFuture, replicaFuture)
-      } else if (primaryFuture != null) {
-        primaryFuture
-      } else if (replicaFuture != null) {
-        replicaFuture
+      if (primaryFuture._1 != null && replicaFuture._1 != null) {
```

Comment:
can you deconstructor the tuple and assign _1 and _2 with a proper name, for better readability?

`val (a, b) = commitFiles ...`

Reply:
Done~